File: src/main/java/com/mrkirby153/snowsgivingbot/entity/SettingEntity.java
package com.mrkirby153.snowsgivingbot.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.NonNull;
import lombok.RequiredArgsConstructor;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.Table;

@Entity
@Table(name = "settings")
@Data
@AllArgsConstructor
@NoArgsConstructor
@RequiredArgsConstructor
public class SettingEntity {

    @Id
    @GeneratedValue
    private long id;

    @NonNull
    @Column(name = "guild")
    private String guild;

    @NonNull
    @Column(name = "`key`")
    private String key;

    @NonNull
    @Column(name = "`value`")
    private String value;
}

File: src/main/kotlin/com/mrkirby153/giveaways/jpa/SettingEntity.kt
package com.mrkirby153.giveaways.jpa

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Table
import org.springframework.data.jpa.repository.JpaRepository

@Entity
@Table(name = "settings")
class SettingEntity(
    val guild: String,
    @Column(name = "`key`")
    val key: String,
    @Column(name = "`value`")
    val value: String
) : AutoIncrementingJpaEntity<Long>() {
}

interface SettingRepository : JpaRepository<SettingEntity, Long> {

}